components/collator/src/elements.rs

```diff
@@ -1477,8 +1443,18 @@ where
                         let offset = usize::from(trail_or_complex & 0xFFF);
                         if offset < self.scalars16.len() {
                             let len = usize::from(trail_or_complex >> 13) + 2;
-                            let (starter, tail) =
-                                split_first_u16(self.scalars16.get_subslice(offset..offset + len));
+                            let (starter, tail) = self
+                                .scalars16
+                                .get_subslice(offset..offset + len)
+                                .and_then(|slice| slice.split_first())
```

Comment:
You could use a method reference instead of a lambda:
```suggestion
                                .and_then(ZeroSlice::split_first)
```